scripts/sync-to-homebrew-tap

```diff
@@ -0,0 +1,208 @@
+  echo "❌ Homebrew sync failed"
+  exit 2
+}
+
+trap fail ERR TERM INT
+
+rm -rf "${DOWNLOAD_DIR}" "${SYNC_DIR}" "${BREW_CONFIG_DIR}"
+mkdir -p "${DOWNLOAD_DIR}" "${SYNC_DIR}" "${BREW_CONFIG_DIR}"
+
+BASE_ASSET_URL="https://github.com/${REPO}/releases/download/${VERSION}/${BINARY_BASE}"
+MAC_HASH=""
+LINUX_HASH=""
+LINUX_ARM64_HASH=""
+
+function hash_file() {
+  local file="${1}"
+  echo "$(openssl dgst -sha256 "${file}" | cut -d' ' -f2 | tr -d '\n')"
+}
+
+for os_arch in "${PLATFORMS[@]}"; do
+    os=$(echo "${os_arch}" | cut -d'/' -f1)
+    arch=$(echo "${os_arch}" | cut -d'/' -f2)
+
+    asset_url="${BASE_ASSET_URL}-${os}-${arch}.tar.gz"
+    asset_file="${BINARY_BASE}-${os}-${arch}.tar.gz"
+    asset_file_path="${DOWNLOAD_DIR}/${asset_file}"
+
+    curl -Lo "${asset_file_path}" "${asset_url}"
+
+    asset_file_size=$(du -k "${asset_file_path}" | cut -f1)
+    if [[ "${asset_file_size}" -lt 100 ]]; then
+      ## If we cannot download and dry-run is set, build it locally
+      if [[ "${DRY_RUN}" -eq 1 ]]; then 
+        ${SCRIPTPATH}/build-binaries -d -v "${VERSION}" -p "${os_arch}"
+        cp "${BUILD_DIR}/bin/${asset_file}" ${asset_file_path}
+      else 
+        echo "❗️${asset_file_path} is empty, skipping"
+        continue
+      fi
+    fi
+
+    if [[ "${os}" == "darwin" ]]; then
+      MAC_HASH=$(hash_file "${asset_file_path}")
+    elif [[ "${os}" == "linux" && "${arch}" == "amd64" ]]; then
+      LINUX_HASH=$(hash_file "${asset_file_path}")
+    elif [[ "${os}" == "linux" && "${arch}" == "arm64" ]]; then
+      LINUX_ARM64_HASH=$(hash_file "${asset_file_path}")
+    fi
+
+done
+
+cat << EOM > "${BREW_CONFIG_DIR}/${BINARY_BASE}.json"
+{
+    "name": "${BINARY_BASE}",
+    "version": "${VERSION_NUM}",
+    "bin": "${BINARY_BASE}",
+    "bottle": {
+        "root_url": "${BASE_ASSET_URL}",
+        "sha256": {
+            "sierra": "${MAC_HASH}",
+            "linux": "${LINUX_HASH}",
+            "linux_arm": "${LINUX_ARM64_HASH}"
+        }
+    }
+}
+EOM
+
+if [[ "${DRY_RUN}" -eq 0 ]]; then 
+  if [[ -z "${MAC_HASH}" ]] && [[ -z "${LINUX_HASH}" ]] && [[ -z "${LINUX_ARM64_HASH}" ]]; then 
+    echo "❌ No hashes were calculated and dry-run is NOT engaged. Bailing out so we don't open a bad PR to the tap."
+    exit 4
+  fi
+  cd "${SYNC_DIR}"
+  gh repo fork $TAP_REPO --clone --remote
+  cd "${FORK_DIR}"
+  git remote set-url origin https://${GITHUB_USERNAME}:${GITHUB_TOKEN}@github.com/${TAP_REPO}.git
+  DEFAULT_BRANCH=$(git rev-parse --abbrev-ref HEAD | tr -d '\n')
+
+  cp "${BREW_CONFIG_DIR}/${BINARY_BASE}.json" "${FORK_DIR}/bottle-configs/${BINARY_BASE}.json"
+
+  ## best effort sync of fork
+  git pull upstream ${DEFAULT_BRANCH} || :
+  git push -u origin ${DEFAULT_BRANCH} || :
+
+  git checkout -b "${BINARY_BASE}-${VERSION}"
+  git add "bottle-configs/${BINARY_BASE}.json"
+  git commit -m "${BINARY_BASE} update to version ${VERSION_NUM}"
+
+  git push -u origin "${BINARY_BASE}-${VERSION}"
+  gh pr create --fill 
```

Comment:
may be its a good idea to add labels here or add to body that this PR is auto-generated ? 
What about assignees? not sure how the team keeps up with new PRs cut on their repo.

Reply:
you can't add labels or assignees unless you have access to the repo. We don't have access yet.  But we can update later if that changes

Reply:
sure.